lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py

```diff
@@ -0,0 +1,429 @@
+        choices:
+            - linux
+            - windows
+        default: linux
+    state:
+        description:
+            - Assert the state of the container instance. Use 'present' to create or update an container instance and 'absent' to delete it.
+        default: present
+        choices:
+            - absent
+            - present
+    ip_address:
+        description:
+            - The IP address type of the container group (default is 'none')
+        choices:
+            - public
+            - none
+        default: None
+    ports:
+        description:
+            - List of ports exposed within the container group.
+    location:
+        description:
+            - Valid azure location. Defaults to location of the resource group.
+        default: resource_group location
+    registry_login_server:
+        description:
+            - The container image registry login server.
+    registry_username:
+        description:
+            - The username to log in container image registry server.
+    registry_password:
+        description:
+            - The password to log in container image registry server.
+    containers:
+        description:
+            - List of containers.
+        suboptions:
+            name:
+                description:
+                    - The name of the container instance.
+                required: true
+            image:
+                description:
+                    - The container image name.
+                required: true
+            memory:
+                description:
+                    - The required memory of the containers in GB.
+                default: 1.5
+            cpu:
+                description:
+                    - The required number of CPU cores of the containers.
+                default: 1
+            ports:
+                description:
+                    - List of ports exposed within the container group.
+    force_update:
+        description:
+            - Force update of existing container instance.
+        type: bool
+        default: False
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+  - name: Create sample container group
+    azure_rm_containerinstance:
+    resource_group: testrg
+    name: mynewcontainergroup
+    os_type: linux
+    ip_address: public
+    ports:
+      - 80
+      - 81
+    containers:
+      - name: mycontainer1
+        image: httpd
+        memory: 1.5
+        ports:
+          - 80
+      - name: mycontainer2
+        image: httpd
+        memory: 1.5
+        ports:
+          - 81
+'''
+RETURN = '''
+state:
+    description: Current state of the azure instance
+    returned: always
+    type: dict
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.containerinstance.models import (ContainerGroup,
```

Comment:
Don't we need to pin the modules on a specific version now to ensure API changes don't break the code?

Reply:
@zikalino Are you able to confirm this, my assumption is that we needed to pin the modules to a specific version.